app/models/data_warehouse_export.rb

```diff
@@ -1,13 +1,19 @@
 class DataWarehouseExport < ApplicationRecord
   EARLIEST_RANGE_FROM = Time.new(2000, 1, 1).utc.freeze
 
-  before_create :set_date_range
+  after_initialize :set_date_range
 
   def run
-    Export::Anything.new(Export::Tasks::Extract.all_relevant(date_range)).run
-    Export::Anything.new(Export::Submissions::Extract.all_relevant(date_range)).run
-    Export::Anything.new(Export::Invoices::Extract.all_relevant(date_range)).run
-    Export::Anything.new(Export::Contracts::Extract.all_relevant(date_range)).run
+    ActiveRecord::Base.transaction do
+      ActiveRecord::Base.connection.execute('LOCK data_warehouse_exports IN ACCESS EXCLUSIVE MODE')
```

Comment:
@rgarner Trying to avoid multiple exports running at once, generating overlapping data. This approach might be incorrect, so I'd appreciate your feedback :)

Reply:
I'm not sure we need to do it explicitly – we only need to specify a lock level at the `.transaction` call. Something like `DataWarehouseExport.transaction(<some_lock_level>)` does it, I think. It's just a question of picking the Rails constant that corresponds to the [Postgres lock level](https://www.postgresql.org/docs/9.5/transaction-iso.html#MVCC-ISOLEVEL-TABLE). Reading...

Reply:
(doesn't work. Trying Things :))